microsoft-edge/devtools-guide-chromium/performance/reference.md

```diff
@@ -0,0 +1,1238 @@
+-->
+
+The DevTools experiment **Performance panel: show postMessage dispatch and handling flows** improves the **Performance** tool's **Main** section to help you quickly identify `postMessage` events and handlers, by distinguishing events that are triggered by the `postMessage` method from other events that are displayed in the **Performance** tool.  This experiment helps you investigate performance issues that are related to posting messages across various threads of an application.
+
+Without this experiment, events that are triggered by dispatching and handling messages between threads of an application appear as generic scripting function-call events.  With this experiment enabled:
+* `postMessage` dispatch events appear as **Schedule postMessage**.
+* `postMessage` handler events appear as **On Message**:
+
+!["Schedule postMessage" events and "On Message" events in the Performance tool](./reference-images/post-message-events.png)
+
+This experiment helps you investigate when a `postMessage` call occurred, and how long the message was queued before the `postMessage` handler starts.  The dispatch events are linked to handler events by initiator arrows that appear when you click on either type of event:
+
+![Arrows linking dispatch events to handler events](./reference-images/post-message-events-arrows.png)
+
+To use this feature, in DevTools, select **Customize and control DevTools** (![the Customize and control DevTools icon](./reference-images/customize-and-control-devtools-icon.png)) > **Settings** > **Experiments**, select the checkbox **Performance panel: show postMessage dispatch and handling flows**, click the **Close** (X) button, and then click the **Reload DevTools** button.
+
+See also:
+* [Performance recording event reference](../performance/performance-reference.md)
+* [postMessage Trace Events demo (rendered)](https://microsoftedge.github.io/Demos/devtools-postmessage-perf-timeline/)
+* [postMessage Trace Events demo (source code)](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-postmessage-perf-timeline)
+
+
+<!-- ====================================================================== -->
+## Analyze rendering performance with the Rendering tool
+<!-- https://developer.chrome.com/docs/devtools/performance/reference#rendering -->
+
+Use the **Rendering** tool to help visualize the rendering performance of your page.  The **Rendering** tool is a **Quick View** tool; if you open it from the **Command Menu**, it opens at the bottom of DevTools.
+
+To open the **Rendering** tool:
+
+1. Right-click a webpage, and then select **Inspect**.  Or, press **Ctrl+Shift+I** (Windows, Linux) or **Command+Option+I** (macOS).  DevTools opens.
+
+1. On the **Activity Bar** (or **Quick View**), click the **More tools** (![More tools icon](./reference-images/more-tools-icon.png)) button, and then select **Rendering**.
+
+   The **Rendering** tool opens:
+
+   ![The Rendering tool](./reference-images/rendering-tool.png)
```

Comment:
Broken image.

Reply:
There are at least 3 other broken images further down in this article.

Reply:
@captainbrosset  pls try:

1. go to internal preview anchor:
https://review.learn.microsoft.com/microsoft-edge/devtools-guide-chromium/performance/reference?branch=pr-en-us-3378#analyze-rendering-performance-with-the-rendering-tool
2. In Edge, **F12** (DevTools) > Right-click **Refresh** > **Empty cache and hard refresh** > **F12**.

Predict the pngs are rendered.
 
Expect that to work, b/c the pngs are rendered at the below, and it is a frequent behavior/ action required.

external preview anchor:
https://github.com/MicrosoftDocs/edge-developer/blob/user/mikehoffms/perf-sync/microsoft-edge/devtools-guide-chromium/performance/reference.md#analyze-rendering-performance-with-the-rendering-tool